Include <xlocale.h> from langinfo.h so that locale_t gets defined.

Include <locale.h> from langinfo.h so that LC_ALL gets defined.

diff -u -r /home/fornwall/lib/android-ndk/sources/android/support/include/langinfo.h ./include/langinfo.h
--- /home/fornwall/lib/android-ndk/sources/android/support/include/langinfo.h	2016-03-03 16:54:04.000000000 -0500
+++ ./include/langinfo.h	2016-03-12 17:23:01.187197445 -0500
@@ -31,6 +31,8 @@
 // __LP64__
 
 #include <nl_types.h>
+#include <xlocale.h>
+#include <locale.h>
 
 #define _NL_ITEM(category,index)  (((category) << 10) | (index))
 
@@ -111,10 +112,8 @@
 extern "C" {
 #endif
 
-#if !defined(__LP64__)
 char *nl_langinfo(nl_item);
 char *nl_langinfo_l(nl_item, locale_t);
-#endif // !__LP64__
 
 #ifdef __cplusplus
 }  // extern "C"
